src/main/java/org/stellar/sdk/SorobanServer.java

```diff
@@ -0,0 +1,344 @@
+package org.stellar.sdk;
+
+import com.google.common.io.BaseEncoding;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.stellar.sdk.requests.ClientIdentificationInterceptor;
+import org.stellar.sdk.requests.ResponseHandler;
+import org.stellar.sdk.requests.sorobanrpc.GetEventsRequest;
+import org.stellar.sdk.requests.sorobanrpc.GetLedgerEntriesRequest;
+import org.stellar.sdk.requests.sorobanrpc.GetTransactionRequest;
+import org.stellar.sdk.requests.sorobanrpc.SendTransactionRequest;
+import org.stellar.sdk.requests.sorobanrpc.SimulateTransactionRequest;
+import org.stellar.sdk.requests.sorobanrpc.SorobanRpcErrorResponse;
+import org.stellar.sdk.requests.sorobanrpc.SorobanRpcRequest;
+import org.stellar.sdk.responses.sorobanrpc.GetEventsResponse;
+import org.stellar.sdk.responses.sorobanrpc.GetHealthResponse;
+import org.stellar.sdk.responses.sorobanrpc.GetLatestLedgerResponse;
+import org.stellar.sdk.responses.sorobanrpc.GetLedgerEntriesResponse;
+import org.stellar.sdk.responses.sorobanrpc.GetNetworkResponse;
+import org.stellar.sdk.responses.sorobanrpc.GetTransactionResponse;
+import org.stellar.sdk.responses.sorobanrpc.SendTransactionResponse;
+import org.stellar.sdk.responses.sorobanrpc.SimulateTransactionResponse;
+import org.stellar.sdk.responses.sorobanrpc.SorobanRpcResponse;
+import org.stellar.sdk.xdr.ContractDataDurability;
+import org.stellar.sdk.xdr.ContractEntryBodyType;
+import org.stellar.sdk.xdr.LedgerEntry;
+import org.stellar.sdk.xdr.LedgerEntryType;
+import org.stellar.sdk.xdr.LedgerKey;
+import org.stellar.sdk.xdr.SCVal;
+import org.stellar.sdk.xdr.SorobanAuthorizationEntry;
+import org.stellar.sdk.xdr.XdrDataInputStream;
+import org.stellar.sdk.xdr.XdrDataOutputStream;
+
+/**
+ * Main class used to connect to the Soroban-RPC instance and exposes an interface for requests to
+ * that instance.
+ */
+public class SorobanServer implements Closeable {
+  private static final int SUBMIT_TRANSACTION_TIMEOUT = 60; // seconds
+  private static final int CONNECT_TIMEOUT = 10; // seconds
+  private final HttpUrl serverURI;
+  private final OkHttpClient httpClient;
```

Comment:
maybe this could be an opportunity to replace OkHttpClient because it's really bloated and it pulls in kotlin as a dependency even though it's not used here at all. 

If you want to evaluate http clients a bit, the other alternatives are apache HttpClient or bump the java jdk level of project to 11 and then can use the jdk's built-in `java.net.http.HttpClient`

Reply:
oh, skip taking that on in here, we should just carve out changing the jdk level and updating library deps to a separate 'tech debt' ticket so it doesn't conflate the effort here, thanks!

https://github.com/stellar/java-stellar-sdk/issues/496

cc @mollykarcher , @tamirms